aten/src/ATen/native/transformers/attention.cpp

```diff
@@ -683,10 +683,15 @@ Tensor scaled_dot_product_attention(
         auto out_lse_softmax = at::_scaled_dot_product_flash_attention(
             query_padded, key_padded, value_padded, dropout_p, is_causal, false /*return_debug_mask*/, og_scale.as_float_unchecked());
         return post_process_flash_output(std::get<0>(out_lse_softmax), og_size);
-      }
+      } else if (query_.device().type() == DeviceType::PrivateUse1) {
+        auto out_lse_softmax = at::_scaled_dot_product_flash_attention(
```

Comment:
cc @drisspg in case you have thoughts on how the decomp for sdpa should work for out-of-tree backends (although calling `_scaled_dot_product_flash_attention` seems strictly more reasonable than the CPU-only call below)

Reply:
@drisspg Could you give me some advice?

Reply:
@drisspg @albanD If you have time, could you please help review this commit?